### PR TITLE
imapnotify: Do not shell-escape password cmd

### PR DESCRIPTION
**Pull request by kirelagin**
### Description

Fix a regression introduced in #1675. (cc @duairc @teto)

tldr: `escapeShellArg` breaks the pipe in `pass some-account | head -n1`.

### Checklist

- [ ] Change is backwards compatible. (**ACTUALLY NOT**)

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Comment by rycee** [comment]
In general I think it is better if the user is not expected to escape the command since different programs may execute the password command in different ways. Indeed, I would recommend avoiding using shell functionality in the command since the program may use `exec()` rather than `system()`.

If the password command relies on shell functionality then I would suggest being explicit about it and use something like
``` nix
passwordCommand = toString (pkgs.writeShellScript "my-passwd-cmd" ''
  pass some-account | head -n1
'');
```

**Comment by kirelagin** [comment]
Well, I mean, I don’t really see how shell-escaping the password command does any good to anyone – what could the user potentially put in there that would require escaping? It just makes things more confusing and inconsistent: for example, `pass ... | head -n1` works perfectly fine with `mbsync`, but not with `imapnotify`.

I don’t have much evidence to cite, but it is my understanding that, within the NixOS/nixpkgs ecosystem, `somethingCommand` has always been taken to mean “a shell command”. And, indeed, “different programs may execute the password command in different ways”, but it is the job of the corresponding modules to make sure it works as expected (e.g. by wrapping it into `writeShellScript` if the program uses `exec`). As a matter of fact, giving arguments to an executable by separating them by spaces is “shell functionality” as well, but it is used right in the examples. Lastly, if it is not a shell command, then, I would say, the option should not be called “command” at all, but rather “exec” or something.

I don’t want to argue and put any pressure on anyone, so I’ll close the PR, but feel free to reopen and merge it in case you change your mind.

**Comment by sand4rt** [comment]
It took a while to understand why it wasn’t working, but I eventually ended up with:

```nix
passwordCommand = toString (
  pkgs.writeShellScript "email-work" ''
    ${pkgs.pass}/bin/pass email/work | ${pkgs.coreutils}/bin/head -n1
  ''
);
```

perhaps there should be a comment in the docs about this?
